### ISSUE TITLE
url filter should detect absolute urls without the protocol

### ISSUE THREAD

**Issue by oscarotero** [11ty/eleventy#1247]
**Is your feature request related to a problem? Please describe.**
I'm using the `url` filter for images in this way:

```njk
<img src="{{ image.src | url }}">
```

But sometimes, the images are external urls. To make them work in http and https contexts, I remove the protocol, so the urls are like this: `//placehold.it/300x100`. Seems like the filter does not detect this and generate the invalid url `/placehold.it/300x100`.

**Describe the solution you'd like**
I can see that the filter can detect http and https protocols (https://github.com/11ty/eleventy/blob/master/src/Filters/Url.js#L8) so I propose adding another condition: `url.indexOf("//") === 0`.

**Comment by denisbrodbeck** [comment]
You could write a wrapper `filter` and use that instead.

```js
// .eleventy.js

const url = require(`@11ty/eleventy/src/Filters/Url`);

module.exports = function (config) {
  config.addFilter(`myurl`, (uri) => {
    if( condition) {
        return myspecialcase;
    }
    return url(uri);
  });

  // Base Config
  return {
    dir: {
      input: `src`,
      output: `dist`,
      includes: `_includes`,
      layouts: `_layouts`,
      data: `_data`,
    },
    templateFormats: [`njk`, `md`, `11ty.js`],
    htmlTemplateEngine: `njk`,
    markdownTemplateEngine: `njk`,
  };
};
```

Then use it instead of `url`:

```html
<img src="{{ image.src | myurl }}">
```

**Comment by oscarotero** [comment]
@denisbrodbeck Yep, this would be a workaround, but I think this is a malfunction of the filter, so I'd rather fix it.

**Comment by denisbrodbeck** [comment]
Sure, fire away with a PR 😄

**Comment by zachleat** [comment]
Thanks for the PR! It’s merged and will ship with 1.0
